clippy_utils/src/qualify_min_const_fn.rs

```diff
@@ -256,20 +263,29 @@ fn check_statement<'tcx>(
 
 fn check_operand<'tcx>(tcx: TyCtxt<'tcx>, operand: &Operand<'tcx>, span: Span, body: &Body<'tcx>) -> McfResult {
     match operand {
-        Operand::Move(place) | Operand::Copy(place) => check_place(tcx, *place, span, body),
+        Operand::Move(place) => check_place(tcx, *place, span, body, true),
+        Operand::Copy(place) => check_place(tcx, *place, span, body, false),
         Operand::Constant(c) => match c.check_static_ptr(tcx) {
             Some(_) => Err((span, "cannot access `static` items in const fn".into())),
             None => Ok(()),
         },
     }
 }
 
-fn check_place<'tcx>(tcx: TyCtxt<'tcx>, place: Place<'tcx>, span: Span, body: &Body<'tcx>) -> McfResult {
+fn check_place<'tcx>(tcx: TyCtxt<'tcx>, place: Place<'tcx>, span: Span, body: &Body<'tcx>, in_move: bool) -> McfResult {
```

Comment:
Note that `in_move` here seems to be necessary here, as otherwise, something like `fn name(&self) -> &String { ... }` will not be linted

Reply:
For moves you'll only need to check if the base of the projection has a non-const `Drop`. Only locals, or fields of a local, can be dropped and if a type is const droppable then all of it's fields will be as well.

Reply:
I think it does this now

Reply:
The only part you need to check for a move is the base of the projection. There's no need to check the type of each projection for const `Drop`.

Reply:
I forgot to remove that one :sweat_smile: Fixed, it only checks the base now